build-tools/packages/build-cli/src/commands/generate/typetests.ts

```diff
@@ -105,50 +105,23 @@ export default class GenerateTypetestsCommand extends PackageCommand<
 		// statements into the type test file, we need to use the previous version name.
 		previousPackageJson.name = previousPackageName;
 
-		const { typesPath: currentTypesPathRelative, levelUsed: currentPackageLevel } =
-			getTypesPathWithFallback(currentPackageJson, level, this.logger, fallbackLevel);
-		const currentTypesPath = path.resolve(path.join(pkg.directory, currentTypesPathRelative));
-		this.verbose(
-			`Found ${currentPackageLevel} type definitions for ${currentPackageJson.name}: ${currentTypesPath}`,
-		);
-
 		const { typesPath: previousTypesPathRelative, levelUsed: previousPackageLevel } =
 			getTypesPathWithFallback(previousPackageJson, level, this.logger, fallbackLevel);
 		const previousTypesPath = path.resolve(
 			path.join(previousBasePath, previousTypesPathRelative),
 		);
 		this.verbose(
-			`Found ${previousPackageLevel} type definitions for ${previousPackageJson.name}: ${previousTypesPath}`,
+			`Found ${previousPackageLevel} type definitions for ${currentPackageJson.name}: ${previousTypesPath}`,
```

Comment:
The package name is expected to match in the old and current versions, so using `current` here is fine? Is that the rationale?

Tangent: what happens if a package changes scopes?

Reply:
> The package name is expected to match in the old and current versions, so using `current` here is fine? Is that the rationale?

I think this was a mistake, actually. I'll look closer and fix if needed, add a comment otherwise.

> Tangent: what happens if a package changes scopes?

This already happens today and isn't really supported. I think you can manually update the previous version reference in package.json, and the generation will work, but the automatic adding/removing of the dependency, and updating the versions - none of that works. Still I think this is an edge case for now. Might be something we need better support for in the future.

Reply:
There is no automatic way to set the correct package or version to test against when a package is renamed, so updating the dependency manually is expected. The test generation however should be able to work just fine.